services/src/main/java/org/fao/geonet/api/cssstyle/CssStyleSettingsService.java

```diff
@@ -275,6 +284,41 @@ private String convertFormNamesToLessVariablesNames(String jsonLessVariables) th
         return output.toString();
     }
 
+    /**
+     * Check the JSON file and Less variables values.
+     *
+     * @param jsonLessVariables the json less variables
+     * @return true if the file is a valid JSON
+     *  and the properties are correct
+     * @throws JSONException the JSON exception
+     */
+    private boolean checkJSONFile(String jsonLessVariables) throws JSONException {
+
+        final JSONObject input = new JSONObject(jsonLessVariables);
+        final Iterator<String> iter = input.keys();
+        while (iter.hasNext()) {
+            final String key = iter.next();
+            if (input.getString(key) != null) {
+
+                if(!key.equals("gnBackgroundImage") && !StringUtils.isEmpty(input.getString(key))) {
+                    Pattern pattern = Pattern.compile("^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$");
+                    Matcher matcher = pattern.matcher(input.getString(key));
+                    if (!matcher.matches()) {
+                        throw new JSONException("Invalid URL property");
```

Comment:
This should be `throw new JSONException("Invalid color value. It must be in the format #RRGGBB.")`